New section kiali working with no istiod (#613)
kiali/kiali.io: content/en/docs/Configuration/no-istiod.md

---
title: "No Istiod Access"

> description: "Kiali working with no access to Istiod"

Reviewer: ```suggestion
description: "Kiali behavior with no access to Istiod (the `/debug` endpoints are not available)"
```

---

## Introduction

> There are different scenarios where Kiali is required to work with no access to the Istio registry:

Reviewer: ```suggestion
Kiali makes use of the Istiod `/debug` endpoints for introspection into the control plane.   If this API is unavailable Kiali continues to perform, but the feature set will be degraded.  The Istio API can be unavailable for various reasons:
```

* When using other Service Mesh solutions that doesn't expose Istiod

> * Different [deployment models](https://istio.io/latest/docs/ops/deployment/deployment-models/#multiple-clusters) where Istiod is not available locally and remotely 

Reviewer: ```suggestion
* The Istio API has been explicitly disabled in the Istio configuration.
* The [deployment model](https://istio.io/latest/docs/ops/deployment/deployment-models/#multiple-clusters) prevents access to the Istio API (firewalls, other networking concerns or limitations).
* The API is configured but for some, potentially unexpected, reason can not be reached by Kiali.
```

## Configuration

> Kiali needs to be set up when the Istio registry is not accesible. This is done with a configuration item, istio_api_enabled.  

Reviewer: ```suggestion
When the Istio API is known to be inaccessible Kiali should be configured via the `istio_api_enabled` configuration item.  
```

By default, istio_api_enabled is true. 

```yaml
# ...
spec:
external_services:
  istio:
    istio_api_enabled: false
# ...
```

## How does it affect Kiali

> When Istio registry is not available, there are some expected changes: 

Reviewer: ```suggestion
When the Istio API is not available there is expected feature degradation in Kiali: 
```

* The control plane metrics won't be available.
* The proxy status won't be available in the workloads details view.

> * The namespace list will be obtained directly from the Kubernetes API, because it won't be possible to use the Istio cache. This could affect slightly the performance.

Reviewer: ```suggestion
* The namespace list will be obtained directly from the Kubernetes API, because it won't be possible to use the Istio cache. This may slightly impact performance.
```

> * The [Istio validations](#a-nameistio_validationsa-istio-validations) and the Kiali validations won't be available. There are, though, some cases where they will be still available.

Reviewer: ```suggestion
* The [Istio validations](#a-nameistio_validationsa-istio-validations) may not be available.
* The Kiali validations will not be available.
```

* The [Istio Registry Services](#a-nameistio_registrya-istio-registry-services) that are not present in the Kubernetes list won't be available.

> * The [Istio Configurations](#a-nameistio_configurationsa-istio-configurations) will be available. This is because the list of Istio configurations is obtained using the Kubernetes API. 

Reviewer: ```suggestion

Note that [Istio Configurations](#a-nameistio_configurationsa-istio-configurations) will be available. This is because the list of Istio configurations is obtained using the Kubernetes API. 
```
